nuitka/tools/quality/auto_format/AutoFormat.py

```diff
@@ -646,7 +665,8 @@ def autoFormatFile(
                     else:
                         copyFile(tmp_filename, filename)
 
-            changed = True
+            if filename != "nuitka/build/static_src/HelpersDictionariesGenerated.c":
```

Comment:
something strange is going on with this file. It says it is changed, when there are no changes.

Reply:
There are whitespace only changes. It plagues me very much myself. I think on GitHub it auto-formats wrong with the clang-format used. I suspect that a new runner must be used or something like that. My Debian Buster also does it wrong. Newest Ubuntu and Windows do it correctly. So bugs in clang-format are very annoying.

Since the files are generated, they very much need auto-format, and cannot just be ignored. I would hate for the state of the file to toggle back and forth depending on who last generated it on what platform. I might have to figure out which clang-format it is, and outright reject updating the file unless it's the correct one, or something like that.

Very bad situation and unresolved. Saying it never changed, I am not sure what the impact is, maybe it then does not get replaced when generating it anew.

Reply:
I just checked, and on my Debian clang-format-8 is used, and does it wrong. On my Ubuntu, a fallback I added at some point is used, namely to use VSCode extension:

```
    # Search Visual Code C++ extension for LLVM path.
    if not clang_format_path:
        for extension_path, extension_filename in listDir(
            os.path.expanduser("~/.vscode-server/extensions")
        ):
            if extension_filename.startswith("ms-vscode.cpptools-"):
                with withEnvironmentPathAdded(
                    "PATH",
                    os.path.join(extension_path, "LLVM/bin"),
                ):
                    clang_format_path = getExecutablePath("clang-format")

                break
```

I guess, that is fine, but we need something like a version check in the end, which makes this code less liberal:

```
    clang_format_path = (
        getExecutablePath("clang-format-12")
        or getExecutablePath("clang-format-12")
        or getExecutablePath("clang-format-11")
        or getExecutablePath("clang-format-10")
        or getExecutablePath("clang-format-9")
        or getExecutablePath("clang-format-8")
        or getExecutablePath("clang-format-7")
        or getExecutablePath("clang-format")
    )
```

Reply:
For developers, I would say that the VS Code should not be the fallback, but the first thing tried. But on GitHub Actions, the extension may not be easy to have, so using installed clang-format of proper version should still be good. Problem remains, current runner, `ubuntu-20.04` doesn't have them. But a newer LTS will probably have it. I will make these changes after merging your stuff here though, as you can see, it's already broken as is.

Reply:
On the latest Linux workers, it is possible to install a specific and recent version of clang-format. c.f. https://packages.ubuntu.com/search?suite=jammy&searchon=names&keywords=clang-format

On the older `focal` workers used in this github actions, the choice of clang-format isnt as good, with clang-format-12 as the highest. c.f. https://packages.ubuntu.com/search?suite=focal&searchon=names&keywords=clang-format

There are packages hosted elsewhere that can be used to install more recent clang-format for `focal`, as I used to do this on C++ projects, but I'd need to dig deeper to find them.

if `clang-format-12` is good enough, it is easy enough to install it almost everywhere, and because it is usually installed as `clang-format-12` binary name, the auto formatter can demand it exists if the PR would contain .c , etc file changes.

Reply:
For the user, Visual Code will be good enough, and actually in case of not finding, I will recommend the user to install and use it, but to at least have it. As you can see, the code is not following clang-format releases very well anyway. I did that because debian packages do not provide `clang-format` which in the light of this thing, is probably understandable.

Mind you, I never noticed I didn't have clang-format installed on my other machine. I tried out v11, which is what I can have on my Debian, and it didn't work any better than the v8. As long as the GitHub action runner has something compatible to what Visual Code extension has (which I suspect is always latest greated), we can do well, and there is always backports I guess.

Reply:
I checked the various debian distros, and indeed they do not have `clang-format-12` :/  They have several others though. https://packages.debian.org/search?suite=sid&searchon=names&keywords=clang-format

And brew has 15, 11 and 8 https://formulae.brew.sh/formula/clang-format

And chocolatey only has LLVM 15 https://community.chocolatey.org/packages?q=clang-format&moderatorQueue=&moderationStatus=all-statuses&prerelease=false&sortOrder=relevance

Seems this is complicated enough that it is worth a separate issue to discuss possible solutions, fallbacks, etc